Add notes for iOS 18 Apple Intelligence (#8661)
braze-inc/braze-docs: _docs/_developer_guide/platform_integration_guides/swift/ios_18.md

### iPhone notifications on macOS

Apple's new [iPhone mirroring](https://www.apple.com/newsroom/2024/06/macos-sequoia-takes-productivity-and-intelligence-on-mac-to-new-heights/) feature allows users to receive iPhone notifications on their macOS devices. Keep in mind, some media types, such as Push Story images and GIFs, are not supported, since they can't be rendered as a macOS notification.

### Apple Intelligence

> [Apple Intelligence](https://developer.apple.com/documentation/Updates/Apple-Intelligence) is now available for devices on iOS 18.1 and later. For Braze, the most relevant feature is notification preview summaries, which uses on-device processing to collapse and generate text summaries for groups of related push notifications from a single app. Summaries can be expanded to view the original contents of each individual notification.

Reviewer: ```suggestion
[Apple Intelligence](https://developer.apple.com/documentation/Updates/Apple-Intelligence) is now available for devices running iOS 18.1 and later.

As a Braze user, the most important feature for you to be aware of is [notification summaries](https://support.apple.com/en-us/108781), which use on-device processing to collapse and generate text summaries for groups of related push notifications from an app. End-users can expand an app's summary to view the original content of each individual notification.
```

> Starting in iOS 18.1, Apple launched the rollout of new [Apple Intelligence features](https://developer.apple.com/documentation/Updates/Apple-Intelligence). Most pertinent to Braze are notification preview summaries. This feature uses on-device processing to collapse and generate text summaries for groups of related push notifications from a single app. Summaries can be expanded to view the original contents of each individual notification.

Reviewer 2: nit:
```suggestion
Starting in iOS 18.1, Apple launched the rollout of new [Apple Intelligence features](https://developer.apple.com/documentation/Updates/Apple-Intelligence). The most pertinent feature to Braze is notification preview summaries, which uses on-device processing to collapse and generate text summaries for groups of related push notifications from a single app. Summaries can be expanded to view the original contents of each individual notification.
```

> Keep in mind, this feature must be manually enabled by the user, meaning you won't have control over the specific behavior or content of your app's notification summary. However, this does not impact analytics or reporting features, such as push-click tracking.

Reviewer: ```suggestion
Keep in mind, this feature must be manually enabled by the user, meaning you won't have control over the specific behavior or content of your app's notification summary. However, this won't impact analytics or reporting features, such as push-click tracking.
```

> Since this is a user-enabled feature in the device settings, developers do not have control over the content or behavior of summaries generated by Apple Intelligence. Additionally, these summaries only affect the display of notification text when collapsed as a group and do not impact analytics reporting features, such as push click tracking.

Reviewer: ```suggestion
Keep in mind, this feature must be manually enabled by the user, meaning you won't have control over the specific behavior or content of your app's notification summary. However, this does not impact analytics or reporting features, such as push-click tracking.
```
